src/local/int_cvx.lean

```diff
@@ -0,0 +1,102 @@
+import data.real.basic
+import linear_algebra.basic
+import linear_algebra.finite_dimensional
+import category_theory.endomorphism
+import topology.algebra.module
+import topology.instances.real
+import analysis.convex.basic
+
+-- This is intended as a Lean roadmap for the `int_cvx` lemma, currently Lemma 1.5 of loops.tex.
+-- Rather than a "human readable" proof of a single lemma,
+-- it is a longer sequence of lemmas, facts, and constructions, for now all sorried.
+
+-- TODO everything in this file could be generalised to arbitrary affine spaces over ℝ.
+variables {V : Type*} [add_comm_group V] [vector_space ℝ V]
+
+/-- The dilation about `c` with scale factor `r`. -/
+def dilation (c : V) (r : ℝ) : V → V := sorry
+
+/-- The dilations about `c` give a multiplicative homomorphism from ℝ to End V. -/
+-- In fact, dilations preserve the affine structure,
+-- so one could strengthen this to `ℝ →* End (Aff.of V)`,
+-- if we define `Aff`.
+def dilations (c : V) : ℝ →* category_theory.End V := sorry
+
+-- TODO The @[simp] lemma that this agrees with `dilation`.
+
+/-- A dilation with nonzero scale factor gives an equivalence. -/
+-- This is an easy consequence of the homomorphism property.
+def dilation.equiv (c : V) (r : ℝ) (h : r ≠ 0) : V ≃ V := sorry
+
+-- TODO The @[simp] lemma that this agrees with dilation.
+
+open_locale big_operators
+
+/-- 
+Given a formula for `x` as a weighted center of mass of a finset `t`,
+we give the formula for `x` as a weighted center of mass of `t` dilated by `r` about `c`,
+plus a multiple of `c`.
+
+We'll shortly specialise this to the case `c = barycenter t`.
+-/
+lemma quux (x : V) (t : finset V) (f : V → ℝ) (h : t.center_mass f id = x) (r : ℝ) (c : V) :
+  t.center_mass (λ x, r⁻¹ * f x) (λ x, dilation c r x) + (∑ z in t, (1 - r⁻¹) * f z) • c = x :=
+sorry
+
+variables [topological_space V]
+  [topological_add_group V] [topological_vector_space ℝ V]
+
+lemma dilation.continuous {c : V} {r : ℝ} : continuous (dilation c r) := sorry
+
+lemma foo {c x : V} {s : set V} (h : is_open s) : ∃ ε : ℝ, ε > 0 ∧ dilation c (1+ε) x ∈ s := sorry
```

Comment:
```suggestion
lemma foo {c x : V} {s : set V} (h : is_open s) (hx : x ∈ s) :  ∃ ε > (0:ℝ), dilation c (1+ε) x ∈ s := sorry
```

Reply:
Jalex is introducing two changes here. Let me emphasize the important one is the missing assumption `hx`.